atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraReloadableKeyValueServiceRuntimeConfig.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.cassandra;
+
+import static com.palantir.logsafe.Preconditions.checkArgument;
+
+import com.palantir.atlasdb.cassandra.CassandraServersConfigs.CassandraServersConfig;
+import com.palantir.atlasdb.spi.DerivedConcurrencyConfig;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.refreshable.Refreshable;
+
+public class CassandraReloadableKeyValueServiceRuntimeConfig extends ForwardingCassandraKeyValueServiceRuntimeConfig
+        implements DerivedConcurrencyConfig {
+
+    private final CassandraKeyValueServiceConfig installConfig;
+    private final CassandraKeyValueServiceRuntimeConfig runtimeConfig;
+
+    private CassandraReloadableKeyValueServiceRuntimeConfig(
+            CassandraKeyValueServiceConfig installConfig, CassandraKeyValueServiceRuntimeConfig runtimeConfig) {
+        this.installConfig = installConfig;
+        this.runtimeConfig = runtimeConfig;
+    }
+
+    static Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> fromConfigs(
+            CassandraKeyValueServiceConfig installConfig,
+            Refreshable<CassandraKeyValueServiceRuntimeConfig> runtimeConfigRefreshable) {
+        return runtimeConfigRefreshable.map(runtimeConfig ->
+                validate(new CassandraReloadableKeyValueServiceRuntimeConfig(installConfig, runtimeConfig)));
+    }
+
+    @Override
+    public CassandraKeyValueServiceRuntimeConfig delegate() {
+        return runtimeConfig;
+    }
+
+    @Override
+    public CassandraServersConfig servers() {
+        return installConfig.servers().orElseGet(runtimeConfig::servers);
+    }
+
+    @Override
+    public int replicationFactor() {
+        return installConfig.replicationFactor().orElseGet(runtimeConfig::replicationFactor);
+    }
+
+    @Override
+    public int concurrentGetRangesThreadPoolSize() {
+        return installConfig
+                .concurrentGetRangesThreadPoolSize()
+                .orElseGet(() -> installConfig.poolSize() * servers().numberOfThriftHosts());
```

Comment:
what if I want to override this in the runtime config? same with below

Reply:
Discussed - https://github.com/palantir/atlasdb/pull/5988#discussion_r842215330 - overriding at the runtime config doesn't make sense, given that these fields are not live reloadable. However, these are now overridable at the install config level and apply to the merged config.